app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -144,120 +160,111 @@ class SuggestedEditsContributionsFragment : Fragment(), SuggestedEditsTypeItem.C
                         if (!qLangMap.containsKey(userContribution.title)) {
                             qLangMap[userContribution.title] = HashSet()
                         }
-                        continuedArticlesContributions.add(Contribution(userContribution.title, "", "", EDIT_TYPE_ARTICLE_DESCRIPTION,
-                                "", DateUtil.iso8601DateParse(userContribution.timestamp), WikiSite.forLanguageCode(descLang), 0, 0))
+                        wikidataContributions.add(Contribution(userContribution.title, "", "", EDIT_TYPE_ARTICLE_DESCRIPTION,
+                                null, DateUtil.iso8601DateParse(userContribution.timestamp), WikiSite.forLanguageCode(descLang), 0))
 
                         qLangMap[userContribution.title]!!.add(descLang)
                     }
                     ServiceFactory.get(WikiSite(Service.WIKIDATA_URL)).getWikidataLabelsAndDescriptions(qLangMap.keys.joinToString("|"))
                             .subscribeOn(Schedulers.io())
-                            .observeOn(AndroidSchedulers.mainThread())
-                }
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe({
-                    for (entityKey in it.entities().keys) {
-                        val entity = it.entities()[entityKey]!!
-                        for (contribution in continuedArticlesContributions) {
-                            if (contribution.qNumber == entityKey) {
-                                contribution.title = entity.labels()[contribution.wikiSite.languageCode()]!!.value()
-                                contribution.description = entity.descriptions()[contribution.wikiSite.languageCode()]!!.value()
+                            .flatMap { entities ->
+                                for (entityKey in entities.entities().keys) {
+                                    val entity = entities.entities()[entityKey]!!
+                                    for (contribution in wikidataContributions) {
+                                        if (contribution.qNumber == entityKey) {
+                                            contribution.title = entity.labels()[contribution.wikiSite.languageCode()]!!.value()
+                                            contribution.description = entity.descriptions()[contribution.wikiSite.languageCode()]!!.value()
+                                        }
+                                    }
+                                }
+                                Observable.just(wikidataContributions)
                             }
-                        }
-                    }
-                }, { t ->
-                    L.e(t)
-                    showError(t)
-                }))
-    }
+                },
+                if (allContributions.isNotEmpty() && imageContributionsContinuation.isNullOrEmpty()) Observable.just(Collections.emptyList<Contribution>()) else
```

Comment:
Explicit type argument can be removed.

Reply:
I don't think it can. Try it for yourself and see if the build succeeds.

Reply:
You are right. How about just use `Observable.empty()`?

Reply:
`Observable.empty()` cannot be used because it causes the subscriber to wait forever, because it's empty.  It needs to be non-empty so that the subscriber receives something.

Reply:
I see. Thanks.